openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -416,6 +408,169 @@ def request_password_change(email, orig_host, is_secure):
         raise UserNotFound
 
 
+def check_username_is_valid(username):
+    """Check whether the username input is valid.
+
+    :param username: The proposed username (unicode).
+    :return: True => if the username passes the validity tests.
+             False => if the username fails to pass the validity tests.
+    """
+    try:
+        _validate_username(username)
+    except (AccountUsernameInvalid, UnicodeError):
+        return False
+    return True
+
+
+def check_username_exists(username):
+    """Check whether the username exists in the system.
+
+    :param username: The proposed username (unicode).
+    :return: True => if the username exists in the system.
+             False => if the username doesn't exist in the system.
+    """
+    try:
+        _validate_username_doesnt_exist(username)
+    except (AccountUsernameAlreadyExists):
+        return True
+    return False
+
+
+def get_username_validation_error(username, default=''):
+    """Get the built-in validation error message for when
+    the username is invalid in some way.
+
+    :param username: The proposed username (unicode).
+    :param default: The message to default to in case of no error.
+    :return: Validation error message.
+    """
+    try:
+        _validate_username(username)
+    except AccountUsernameInvalid as invalid_username_err:
+        return str(invalid_username_err)
+    return default
+
+
+def get_username_existence_validation_error(username, default=''):
+    """Get the built-in validation error message for when
+    the username has an existence conflict.
+
+    :param username: The proposed username (unicode).
+    :param default: The message to default to in case of no error.
+    :return: Validation error message.
+    """
+    try:
+        _validate_username_doesnt_exist(username)
+    except AccountUsernameAlreadyExists as username_exists_err:
+        return str(username_exists_err)
+    return default
```

Comment:
I wanted to make existence checks separate because it's not always expected that one gets a response based off of existence when checking for whether the string is "valid".